ToolBox/QuiverDataDownloader/QuiverDataDownloader.cs

```diff
@@ -0,0 +1,246 @@
+                        }
+
+                        if (response.StatusCode == HttpStatusCode.Unauthorized)
+                        {
+                            var finalRequestUri = response.RequestMessage.RequestUri; // contains the final location after following the redirect.
+                            response = client.GetAsync(finalRequestUri).Result; // Reissue the request. The DefaultRequestHeaders configured on the client will be used, so we don't have to set them again.
+
+                        }
+
+                        response.EnsureSuccessStatusCode();
+
+                        return await response.Content.ReadAsStringAsync();
+                    }
+                }
+                catch (Exception e)
+                {
+                    Log.Error(e, $"QuiverDataDownloader.HttpRequester(): Error at HttpRequester. (retry {retries}/{_maxRetries})");
+                    Thread.Sleep(1000);
+                }
+            }
+
+            throw new Exception($"Request failed with no more retries remaining (retry {_maxRetries}/{_maxRetries})");
+        }
+
+        /// <summary>
+        /// Saves contents to disk, deleting existing zip files
+        /// </summary>
+        /// <param name="destinationFolder">Final destination of the data</param>
+        /// <param name="ticker">Stock ticker</param>
+        /// <param name="contents">Contents to write</param>
+        protected void SaveContentToFile(string destinationFolder, string ticker, IEnumerable<string> contents)
+        {
+            ticker = ticker.ToLowerInvariant();
+            var bkPath = Path.Combine(destinationFolder+"/bk/", $"{ticker}.csv");
+            var finalPath = Path.Combine(destinationFolder, $"{ticker}.csv");
+            var finalFileExists = File.Exists(finalPath);
+
+            var lines = new HashSet<string>(contents);
+            if (finalFileExists)
+            {
+                Log.Trace($"QuiverDataDownloader.SaveContentToZipFile(): Adding to existing file: {finalPath}");
+                foreach (var line in File.ReadAllLines(finalPath))
+                {
+                    lines.Add(line);
+                }
+            }
+            else
+            {
+                Log.Trace($"QuiverDataDownloader.SaveContentToZipFile(): Writing to file: {finalPath}");
+            }
+
+            var finalLines = lines.OrderBy(x => DateTime.ParseExact(x.Split(',').First(), "yyyyMMdd", CultureInfo.InvariantCulture, DateTimeStyles.AdjustToUniversal))
+                .ToList();
+
+            var tempPath = Path.Combine(Path.GetTempPath(), $"{Guid.NewGuid()}.tmp");
+            File.WriteAllLines(tempPath, finalLines);
+            var tempFilePath = new FileInfo(tempPath);
+            if (finalFileExists)
+            {
+                tempFilePath.Replace(finalPath,bkPath);
+            }
+            else
+            {
+                tempFilePath.MoveTo(finalPath);
+            }
```

Comment:
Almost there! Let's clean up the backup path resources we left behind once we're done moving the new data into the final path. Good one 👍